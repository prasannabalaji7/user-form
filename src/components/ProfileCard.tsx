import React, { useState } from 'react';
import classNames from 'classnames';
import { Card, Image, ListGroupItem, ListGroup } from 'react-bootstrap';
import Popup from './Popup';
import { popupMessage, imgUrl } from '../constants/Constants';

export interface ProfileProps {
	userProfileName: string;
	userProfileRole: string;
	userProfileCountry: string;
	isEditBtnVisible: boolean;
}

const ProfileCard: React.FC<ProfileProps> = (props) => {
	const [popup, setPopup] = useState(popupMessage);
	const { isEditBtnVisible, userProfileName, userProfileCountry, userProfileRole } = props;
	const editableClass = classNames('childContainer', {
		profileReadOnly: isEditBtnVisible,
	});

	return (
		<div className={editableClass}>
			<Card className='card text-center w-75'>
				<Card.Body>
					<Image
						id='profilepic'
						src={imgUrl}
						alt='laugh'
						roundedCircle
					/>
					<ListGroup className='list-group-flush borderless'>
						<ListGroupItem className='borderless h5'>
							{userProfileName}
						</ListGroupItem>
						<ListGroupItem className='borderless'>
							{userProfileRole}
						</ListGroupItem>
						<ListGroupItem className='borderless'>
							{userProfileCountry}
						</ListGroupItem>
					</ListGroup>
				</Card.Body>
			</Card>
			<Popup
				show={popup.modalShow}
				onHide={() =>
					setPopup({ modalShow: false, title: '', message: '' })
				}
				title={popup.title}
				message={popup.message}
			/>
		</div>
	);
};

export default ProfileCard;