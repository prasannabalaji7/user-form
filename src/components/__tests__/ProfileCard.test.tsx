import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import ProfileCard from '../ProfileCard';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

describe('<ProfileCard />', () => {
    test('should display the values', async () => {
        const { getByText } = render(
            <ProfileCard
                userProfileName='Test123'
                userProfileRole='TestRole'
                userProfileCountry='Singapore'
                isEditBtnVisible={false}
            />
        );
        expect(getByText('Test123')).toBeInTheDocument();
        expect(getByText('TestRole')).toBeInTheDocument();
        expect(getByText('Singapore')).toBeInTheDocument();
    });
    test('Snapshot Testing', () => {
        const tree = renderer
            .create(
                <ProfileCard
                    userProfileName='Test123'
                    userProfileRole='TestRole'
                    userProfileCountry='Singapore'
                    isEditBtnVisible={false}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});