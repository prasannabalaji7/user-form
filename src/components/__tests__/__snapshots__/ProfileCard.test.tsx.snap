// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileCard /> Snapshot Testing 1`] = `
<div
  className="childContainer"
>
  <div
    className="card text-center w-75 card"
  >
    <div
      className="card-body"
    >
      <img
        alt="laugh"
        className="rounded-circle"
        id="profilepic"
        src="https://i.stack.imgur.com/YQu5k.png"
      />
      <div
        className="list-group-flush borderless list-group"
        onKeyDown={[Function]}
      >
        <div
          className="borderless h5 list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Updated Name
        </div>
        <div
          className="borderless list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Updated Role
        </div>
        <div
          className="borderless list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Singapore
        </div>
      </div>
    </div>
  </div>
</div>
`;
